launcher/src/main/java/org/apache/spark/launcher/SparkSubmitOptionParser.java

```diff
@@ -76,6 +76,12 @@
   protected final String PRINCIPAL = "--principal";
   protected final String QUEUE = "--queue";
 
+  // Kubernetes-only options.
+  protected final String KUBERNETES_MASTER = "--kubernetes-master";
+  protected final String KUBERNETES_NAMESPACE = "--kubernetes-namespace";
+  protected final String KUBERNETES_UPLOAD_JARS = "--upload-jars";
```

Comment:
Correct. The API here is quite difficult to define. One could merge this in with --jars, for instance. However, what if the user wants the files in --jars to be interpreted as _files on the driver docker image_? I'm not sure what the best API is for separating user-local jars vs. jars on the docker container's disk.